website/project/views/node.py

```diff
@@ -735,6 +735,14 @@ def _view_project(node, auth, primary=False,
             'institutions': get_affiliated_institutions(node) if node else [],
             'has_draft_registrations': node.has_active_draft_registrations,
             'is_preprint': node.is_preprint,
+            'has_moderated_preprint': node.preprints.filter(provider__reviews_workflow__isnull=False).exists(),
+            'preprint_state': node.preprints.get_queryset()[0].reviews_state
+            if node.preprints.filter(provider__reviews_workflow__isnull=False).exists() else '',
+            'preprint_word': node.preprints.get_queryset()[0].provider.preprint_word if node.is_preprint else '',
+            'preprint_provider': {
+                'name': node.preprints.get_queryset()[0].provider.name,
+                'workflow': node.preprints.get_queryset()[0].provider.reviews_workflow
```

Comment:
Instead of repeating this query over and over, it might be better to add a method/property to the node to get the node's preprint. You could call it once and use the result several times.

It might make sense to move the logic you added to `preprint_url` there -- first try to get a published preprint, then fall back to any preprint -- and use the new method in `preprint_url`